### PR TITLE
writeup overview

### LINKED ISSUE #12: bootcamp section writeup
# overview of project
- [ x ] crud operation
- [ x ] provider schema
- - link to documentation on schema
- [ x ] boiler plate hierarchy
- - explane files within internal/provider
- - snippets are good

### PR DESCRIPTION
**Pull request by snowmang1**
- add writeup to docs
- added crud and schema overview to writeup

### CODE REVIEW

File: writeup/overview.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+### Create.Read.Update.Delete (CRUD)
+Terraform providers work closely with the idea of CRUD operations. Meaning that the providers
+try to classify API functionality into 4 groups, here are examples related to our API:
+- C[reate]
+- - Creates new resource; dev, ops, devops
```
Comment (baserrato) [inline writeup/overview.md]: Line 6: engineer is also a resource

**Review by MinecraftSt3v3 — COMMENTED** [review]
maybe we should add a high level overview of our file structure (snapshot), and explain the different files we had to update, add, and change. as well as what a resource is and data source

File: writeup/overview.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- R[ead]
+- - Read in an existing resource
+- U[pdate]
+- - Update an existing resource, however we can not effect the sub-resources (engineers) inside it
```
Comment (baserrato) [inline writeup/overview.md]: Line 10: ~effect~ affect

File: writeup/overview.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- D[elete]
+- - Deletes an existing resource, however we can not effect those resource's children
```
Comment (baserrato) [inline writeup/overview.md]: Line 12: ~effect~ affect

**Comment by snowmang1** [comment]
that might be adding to much, however  I like the idea so maybe we can describe the different file groups we have.
ex:
make all structures in a single place (modules) and reference them from there

File: writeup/overview.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+![boiler plate file tree](file_tree.png)
+### Look inside the internal directory
+``` Go
+    type ExampleDataSourceModel struct {
+        ConfigurableAttribute types.String `tfsdk:"configurable_attribute"`
+        Id                    types.String `tfsdk:"id"`
+    }
+```
+- this code can be found [example_data_source.go](https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/95358ccbb8c897150c9cbe973c237448bdefcd4f/internal/provider/example_data_source.go)
+- this is how data should be modled and stored in the provider
```
Comment (baserrato) [inline writeup/overview.md]: Line 28: ~modled~ modeled

File: writeup/overview.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#### [example_resource.go](https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/95358ccbb8c897150c9cbe973c237448bdefcd4f/internal/provider/example_resource.go)
+- This files shows use of crud operations
+- As well as metadata and GetSchema which are important for documentation purposes
```
Comment (baserrato) [inline writeup/overview.md]: Line 31 Maybe something like: As well as metadata and GetSchema which are important for building a resource block in terraform.

**Review by baserrato — APPROVED** [review]

**Pull request merged**